### PR TITLE
kvcoord: Fix observability data race

### PR DESCRIPTION
**Pull request by miretskiy**
Fix data race in management of `InCatchup` range
information.

Epic: None

Release note: None

### CODE REVIEW

**Review by erikgrinaker — COMMENTED** [review]
Isn't this going to change the meaning of the metric? Previously, it only reported being in a catchup scan after it had acquired the semaphore and was actually running the catchup scan. Now, it will report being in a catchup scan even before starting the scan. Is that what we want?

**Comment by miretskiy** [comment]
> Isn't this going to change the meaning of the metric? Previously, it only reported being in a catchup scan after it had acquired the semaphore and was actually running the catchup scan. Now, it will report being in a catchup scan even before starting the scan. Is that what we want?

I don't think it's that important, but easy enough to fix.

File: pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go [hunk 1]
```diff
@@ -520,6 +515,7 @@ func newActiveRangeFeed(
 			Span:        span,
 			StartAfter:  startAfter,
 			CreatedTime: timeutil.Now(),
+			InCatchup:   true,
```
Comment (erikgrinaker) [inline pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:518]: I don't think we should set this anymore?

Comment (miretskiy) [reply pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go]: ack.  Thanks.

**Review by erikgrinaker — APPROVED** [review]

**Pull request merged**